docs/how_it_works.md

```diff
@@ -0,0 +1,142 @@
+  the given expression
+- `DataFrame.filter`: evaluate the given expression, and if it only returns a single Series, then
+  only keep rows where the result is `True`.
+
+Now let's turn our attention to the implementation.
+
+## Polars implementation
+
+For Polars, Narwhals just "passes everything through". For example consider the following:
+```python
+import polars as pl
+import narwhals as nw
+
+df_pl = pl.DataFrame({'a': [1,2,3], 'b': [4,5,6]})
+df = nw.from_native(df_pl)
+df.select(nw.col('a')+1)
+```
+
+`nw.col('a')` produces a `narwhals.expression.Expr` object, which has a private `_call` method.
+We can call `nw.col('a')._call(pl)`, then the result is actually `pl.col('a')`.
+
+We then let Polars do its thing. Which is nice, but also not particularly interesting.
+How about translating expressions to pandas? Well, it's
+interesting to us, and you're still reading, so maybe it'll be interesting to you too.
+
+## pandas implementation
+
+When we called `nw.col('a')._call(pl)`, we got a Narwhals-compliant Polars namespace.
+The pandas namespace (`pd`) isn't Narwhals-compliant, as the pandas API is very different
+from Polars'. So...Narwhals implements a `PandasNamespace`, which includes the top-level
+Polars functions included in the Narwhals API:
+
+```python
+import narwhals as nw
+from narwhals._pandas_like.namespace import PandasNamespace
+
+pn = PandasNamespace(implementation='pandas')
+nw.col('a')._call(pn)
+```
+The result from the last line above is the same as we'd get from `pn.col('a')`, and it's
+a `narwhals._pandas_like.expression.PandasExpr` object, which we'll call `PandasExpr` for
+short.
+
+`PandasExpr` also have a `_call` method - but this one expects a `PandasDataFrame` as input.
+Recall from above that an expression is a function from a dataframe to a sequence of series.
+The `_call` method gives us that function! Let's see it in action.
+
+Note: the following examples uses `PandasDataFrame` and `PandasSeries`. These are wrappers
+around pandas DataFrame and pandas Series, which are Narwhals-compliant. To get the native
+pandas objects out from inside them, we access `PandasDataFrame._dataframe` and `PandasSeries._series`.
+
+```python
+import narwhals as nw
+from narwhals._pandas_like.namespace import PandasNamespace
+from narwhals._pandas_like.dataframe import PandasDataFrame
+import pandas as pd
+
+pn = PandasNamespace(implementation='pandas')
+
+df_pd = pd.DataFrame({'a': [1,2,3], 'b': [4,5,6]})
+df = PandasDataFrame(df_pd, implementation='pandas')
+expression = pn.col('a') + 1
+result = expression._call(df)
+print([x._series for x in result])
+```
+The first (and only) Series to be output is:
+```
+0    2
+1    3
+2    4
+Name: a, dtype: int64
+```
+
+So indeed, our expression did what it said on the tin - it took some dataframe, took
+column 'a', and added 1 to it.
+
+If you search for `def register_expression_call`, you'll see that that's all
```

Comment:
Let's handle the link ourselves:

```suggestion
If you search for [`register_expression_call`](https://github.com/narwhals-dev/narwhals/blob/f18c533a0c2ff870ad489a6f93bca8bef2a98eda/narwhals/_pandas_like/utils.py#L171){target="_blank"}, you'll see that that's all
```

Reply:
the `{target="_black"}` didn't work for me when I rendered this. in any case, are sure? I think the link would quickly go out-of-date

Reply:
The one I added is the permalink, not just the link to main and line, which in theory should be more stable to changes in the file. `{target="_blank"}` should open on a new tab/page, which is something I prefer but can live without 😂

Reply:
do i need to enable some extra feature for it to work? if i copy and paste your snippet, i get:

![image](https://github.com/narwhals-dev/narwhals/assets/33491632/ebfe6460-2102-46d8-b013-8e6b20acf0da)

Reply:
I will take a look, but if you want to merge it, let's remove it for now 😇